cli/src/pcluster/config/cluster_config.py

```diff
@@ -1629,17 +1628,85 @@ def __init__(
         self.min_count = Resource.init_param(min_count, default=DEFAULT_MIN_COUNT)
         self.spot_price = Resource.init_param(spot_price)
         self.disable_simultaneous_multithreading = Resource.init_param(
             disable_simultaneous_multithreading, default=False
         )
-        self.__instance_type_info = None
         self.efa = efa or Efa(enabled=False, implied=True)
         self.schedulable_memory = Resource.init_param(schedulable_memory)
 
+
+class FlexibleInstanceType(Resource):
+    """Represent an instance type listed under InstanceTypes under ComputeResources."""
+
+    def __init__(self, instance_type: str, **kwargs):
+        super().__init__(**kwargs)
+        self.instance_type = Resource.init_param(instance_type)
+
+
+def get_instance_type_info(instance_type) -> InstanceTypeInfo:
+    """Return instance type information."""
+    return AWSApi.instance().ec2.get_instance_type_info(instance_type)
+
+
+class SlurmFlexibleComputeResource(BaseSlurmComputeResource):
+    """Represents a Slurm Compute Resource with Multiple Instance Types."""
+
+    def __init__(self, instance_types: List[FlexibleInstanceType], **kwargs):
+        super().__init__(**kwargs)
+        self.instance_types = Resource.init_param(instance_types)
+
+    @property
+    def instance_type_names(self) -> List[str]:
+        """Return list of instance type names in this compute resource."""
+        return [flexible_instance_type.instance_type for flexible_instance_type in self.instance_types]
+
+    @property
+    def disable_simultaneous_multithreading_manually(self) -> bool:
+        """Return true if simultaneous multithreading must be disabled with a cookbook script."""
+        return self.disable_simultaneous_multithreading
+
+    @property
+    def disable_simultaneous_multithreading_via_cpu_options(self) -> bool:
+        """CPU Options not supported for Compute Resources with Flexible Instance Types."""
+        return False
+
+    @property
+    def max_network_interface_count(self) -> int:
+        """Return max number of NICs for the compute resource.
+
+        In this case the Compute Resource may have multiple instance types, hence the instance-type with
+        the least MaxNetworkInterfaceCards value will be considered.
+        """
+        least_max_nics = get_instance_type_info(self.instance_type_names[0]).max_network_interface_count()
+        if len(self.instance_type_names) > 1:
+            for instance_type in self.instance_type_names[1:]:
+                instance_type_info = get_instance_type_info(instance_type)
+                max_nics = instance_type_info.max_network_interface_count()
+                if max_nics < least_max_nics:
+                    least_max_nics = max_nics
+        return least_max_nics
+
+    @property
+    def is_ebs_optimized(self) -> bool:
+        """Return True if all the instances have optimized EBS support."""
+        for instance_type in self.instance_type_names:
```

Comment:
I think [all](https://docs.python.org/3/library/functions.html#all) is what would work here.
I went for this approach to avoid trying to build an iterable/list in order to use `all`. 
But there might be a possibility to use a generator (or generator comprehension). I'll give it a try to see the results.

Reply:
Eventually decided to avoid using `all` here.
Using `all` works but it's not easily readable.